src/coprocessor/statistics/analyze.rs

```diff
@@ -108,15 +133,25 @@ impl<S: Snapshot> AnalyzeContext<S> {
 impl<S: Snapshot> RequestHandler for AnalyzeContext<S> {
     async fn handle_request(&mut self) -> Result<Response> {
         let ret = match self.req.get_tp() {
-            AnalyzeType::TypeIndex => {
+            AnalyzeType::TypeIndex | AnalyzeType::TypeCommonHandle => {
                 let req = self.req.take_idx_req();
-                let mut scanner = ScanExecutor::index_scan_with_cols_len(
-                    EvalContext::default(),
-                    i64::from(req.get_num_columns()),
-                    mem::replace(&mut self.ranges, Vec::new()),
-                    self.storage.take().unwrap(),
-                )?;
-                let res = AnalyzeContext::handle_index(req, &mut scanner);
+                let ranges = mem::replace(&mut self.ranges, vec![]);
+                table::check_table_ranges(&ranges)?;
+                let mut scanner = RangesScanner::new(RangesScannerOptions {
```

Comment:
Is it possible to be all RangesScanner or all BatchScanExecutor?

Reply:
For `AnalyzeIndexReq`, the algorithm prefers to process each row, however `BatchResult` is a column base sub-table which is not suitable for processing in a row-base manner.